front/components/assistant/conversation/ConversationContainer.tsx

```diff
@@ -72,12 +72,39 @@ export function ConversationContainer({
     limit: 50,
   });
 
+  const { setSelectedAssistant } = useContext(InputBarContext);
+
+  const setInputbarMention = useCallback(
+    (agentSid: string) => {
+      setSelectedAssistant({ configurationId: agentSid });
+      setAnimate(true);
+    },
+    [setAnimate, setSelectedAssistant]
+  );
+
   useEffect(() => {
     if (animate) {
       setTimeout(() => setAnimate(false), 500);
     }
   });
 
+  useEffect(() => {
```

Comment:
We already monitor the router within `AssistantConversation`, so this logic should reside there I guess and ensure it only inject the assistant id when `cId` is set to `new`.